test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/delegatingclients/DelegatingSyncClientTest.java

```diff
@@ -45,7 +45,7 @@ public class DelegatingSyncClientTest {
 
     private static final String INTERCEPTED_HEADER = "intercepted-header";
     private static final String INTERCEPTED_HEADER_VALUE = "intercepted-value";
-    private static final String RESPONSE = "response";
+    private static final String RESPONSE = "{\"key\":\"value\"}";
```

Comment:
Just curious, why do we need to fix this?

Reply:
Discussed offline, it's because `JsonParseException` is an IOException and it gets caught here https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/FunctionalUtils.java#L110 and gets converted to UncheckedIOException.

We probably don't want to retry JsonParseException since the request will likely fail again due to invalid response.

Alternatively, we can make changes in InputStreamSubscriber to catch UncheckedIOException and then throw it as IOException in read methods. This is in the contract of InputStream, although it also has backwards compatibility concerns.

Let's discuss with the team.